src/Arcus.Messaging.Abstractions/MessageHandling/MessageProcessingResult.cs

```diff
@@ -1,24 +1,49 @@
 ﻿using System;
-using GuardNet;
 
 namespace Arcus.Messaging.Abstractions.MessageHandling
 {
+    /// <summary>
+    /// Represents all the possible errors of a <see cref="MessageProcessingResult"/>.
+    /// </summary>
+    public enum MessageProcessingError
+    {
+        /// <summary>
+        /// Defines an error that shows that the message processing was interrupted by some external cause,
+        /// unrelated to the message routing.
+        /// </summary>
+        ProcessingInterrupted,
```

Comment:
We will run into the `ProcessingInterrupted` error when:
* The message router has an unexpected exception during message routing. We don't throw exceptions anymore, but there was an catch block that abandons the message in case of such unexpected exception. Maybe we should remove this catch block, as I believe that this is not expected behavior.
* Same goes for the message pump, there was a catch block arround the single message processing. Maybe this can also be removed.
* Then we have two references in the message pump where we `null` check a message and when the pump is cancelled.

> Exception thrown by messagehandler vs messagehandler returning 'no success' ?

This is the same thing. The message handler does not return anything, but if it throws, the `MatchedHandlerFailed` error will be used.

Reply:
- An unexpected exception during message routing is something else then a message for which the MessageRouter can't find a MessageHandler ?

- The catch block around the single message processing: does this catch the exceptions that might be thrown by the MessageHandler's implementation  ?  It seems that this is already handled somewhere else.   If so, I think we can indeed remove that catch block.

Reply:
Exceptions locations will be handled separately in a diff session.

Reply:
Will be discussed later in #456